### DISCUSSION TITLE
Best way to handle backend api calls starting with `/api` route

### DISCUSSION THREAD

**Discussion by kvedantmahajan** [vercel/next.js#19908] [Help]
`next.js` gives default `/api` folder to build node based apis. If the backend is on different stack which also mounts the calls on `/api` which usually is the case, what is the best way to go about it? By default, the requests go to `/api` folder. I strongly feel that there is a missing documentation for this part. Leaving open to the internet to describe best ways which normally involves setting up custom server and proxy middlewares.

I wish to avoid setting up express instance altogether to stick to many defaults of next.js and also to make things simple. 

So, shall we define a dedicated path under api folder and then make a external server call inside that individual `xy-api.js` handler file for each of the application api? 
or use some form of proxy to bypass the default `/api` nextjs route path to make a call directly to the back-end server?

Anyway which way call is origination from nextjs (server side ). Is there a downside to performance for choosing one over the other. 
For instance, server side calls will be made inside `getServerSideProps` method. I am using a state management library with a middleware -  `redux-saga`.    Essentially I am dispatching actions inside this method. So, from above two methods, which one is a better way to go about it if my back-end Python/Go/Java etc.

**Comment by leerob** [comment]
You could use rewrites to forward traffic from `/new-api/*` to `/api`.
